tests/test_cargo_compile_feature_deps.rs

```diff
@@ -0,0 +1,104 @@
+use support::{project, execs};
+use hamcrest::{assert_that, existing_file, not};
+
+fn setup() {
+}
+
+test!(compile_simple_feature_deps {
+    let p = project("foo")
+        .file("Cargo.toml", r#"
+            [project]
+            name = "foo"
+            version = "0.0.1"
+            authors = []
+
+            [features]
+            default = ["a"]
+            a = []
+
+            [[bin]]
+            name = "foo"
+            features = ["a"]
+        "#)
+        .file("src/main.rs", "fn main() {}");
+
+    assert_that(p.cargo_process("build"),
+                execs().with_status(0));
+
+    assert_that(&p.bin("foo"), existing_file());
+
+    assert_that(p.cargo_process("build").arg("--no-default-features"),
```

Comment:
Currently the `cargo_process` helper will blow away the directory and rebuild it, but some platforms (notably windows) don't like blowing away directories right after Cargo uses them, so could this change to just `p.cargo`? That'll avoid blowing away the directory and rebuilding, but should do basically the same thing.